src/edu.umn.cs.melt.copper.compiletime/src/main/java/edu/umn/cs/melt/copper/compiletime/skins/cup/CupSkinParser.java

```diff
@@ -858,8 +858,9 @@ public java.lang.Object runSemanticAction_124()
 	                	}
 	                }
 	                if(terminalFlags.prefix != null) terminal.setPrefix(terminalFlags.prefix);
-	                
-	                if(ignore) ignoreTerminals.add(CopperElementReference.ref(newTerminalName,name.location));
+
+	                //TODO was there a reason this didn't have a current grammar attached?
+	                if(ignore) ignoreTerminals.add(CopperElementReference.ref(currentGrammar.getName(),newTerminalName,name.location));
```

Comment:
The CUP skin only supports a single grammar, so putting a grammar in the reference was not strictly necessary. However, since this is a generated file, it should be kept in sync with its source, [CupSkinGrammar.xml](https://github.com/melt-umn/copper/blob/develop/doc/specs/CupSkinGrammar.xml).

Reply:
Reverted.